### PR TITLE
Distributed processes

### PR DESCRIPTION
**Pull request by oskardudycz**
Added example of distributed processes management using Group Guests Checkout as an example.

It was modelled and explained in detail in the [Implementing Distributed Processes Webinar](https://www.architecture-weekly.com/p/webinar-3-implementing-distributed):

<a href="https://www.architecture-weekly.com/p/webinar-3-implementing-distributed" target="_blank">How to deal with privacy and GDPR in Event-Sourced systems</a>

It shows how to:
- orchestrate and coordinate business workflow spanning across multiple aggregates using [Saga pattern](https://event-driven.io/en/saga_process_manager_distributed_transactions/),
- handle distributed processing both for asynchronous commands scheduling and events publishing,
- getting at-least-once delivery guarantee,
- implementing command store and outbox pattern on top of Marten and EventStoreDB,
- unit testing aggregates and Saga with a little help from [Ogooreck](https://github.com/oskardudycz/Ogooreck),
- testing asynchronous code,

Read more in:
- [Saga and Process Manager - distributed processes in practice](https://event-driven.io/en/saga_process_manager_distributed_transactions/)
- [Event-driven distributed processes by example](https://event-driven.io/en/saga_process_manager_distributed_transactions/).

### CODE REVIEW

File: Sample/HotelManagement/HotelManagement/Core/Structures/Result.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        throw new Exception("That should never happen!");
+    }
+}
+
+public static class Result
+{
+    public static Result<TSuccess, TError> Success<TSuccess, TError>(TSuccess success) => new(success);
+
+    public static Result<TSuccess, TError> Failure<TSuccess, TError>(TError success) => new(success);
```
Comment (Ustingit) [inline Sample/HotelManagement/HotelManagement/Core/Structures/Result.cs:68]: public static Result<TSuccess, TError> Failure<TSuccess, TError>(TError error) => new(error);

File: Sample/HotelManagement/HotelManagement/GuestStayAccounts/GuestStayAccount.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    public Result<GuestCheckedOut, GuestCheckoutFailed> CheckOut(DateTimeOffset now, Guid? groupCheckoutId = null)
+    {
+        if (Status != GuestStayAccountStatus.Opened)
+            throw new InvalidOperationException("Cannot record payment for not opened account");
+
+        return IsSettled
+            ? Success<GuestCheckedOut, GuestCheckoutFailed>(
+                new GuestCheckedOut(
+                    Id,
+                    DateTimeOffset.UtcNow,
```
Comment (Ustingit) [inline Sample/HotelManagement/HotelManagement/GuestStayAccounts/GuestStayAccount.cs:80]: in other places it's common one, not UTC. Just wondering

Comment (oskardudycz) [reply Sample/HotelManagement/HotelManagement/GuestStayAccounts/GuestStayAccount.cs:78]: Should be UTC in all those places, as it's a timestamp of what has happened in the background. Fixed in https://github.com/oskardudycz/EventSourcing.NetCore/pull/167/commits/e93f1412470be074658a3324656bee4e92c0a952. You can read more about my perspective in https://event-driven.io/en/is_keeping_utc_dates_best_solution/.

File: Sample/HotelManagement/HotelManagement/Core/Structures/Result.cs (hunk 1 above)
Comment (oskardudycz) [reply Sample/HotelManagement/HotelManagement/Core/Structures/Result.cs]: Good catch, fixed in https://github.com/oskardudycz/EventSourcing.NetCore/pull/167/commits/e93f1412470be074658a3324656bee4e92c0a952.

**Pull request merged**